### ISSUE TITLE
Fix `no-descending-specificity` false positives for functional pseudo classes

### ISSUE THREAD

**Issue by milkyfuckinway** [stylelint/stylelint#8287]
### What minimal example or steps are needed to reproduce the bug?

```css
.button {
  all: unset;

  &:global(.red) {
    color: rgb(255 0 0);

    &:hover {
      color: rgb(255 100 75);
    }

    &:focus {
      box-shadow: 0 0 0 2px rgb(255 100 75);
    }
  }

  &:global(.blue) {
    color: rgb(50 0 255);

    &:hover {
      color: rgb(50 100 255);
    }

    &:focus {
      box-shadow: 0 0 0 2px rgb(50 100 255);
    }
  }
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "no-descending-specificity": true
  }
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0AjArgFwxAdgAmAB188BDAG3PjzRwGcYMBuY4vPAMngHNyIUKACiQAnGFACUBNuzyQ+I6iO4pBAJgCsGvAAZdEljhnsuACwgA3GCOklZciAqUr1WvAEYdegOwaDxvABfVjtOeAAzCDA0Olt7dhQIAA8AWjpTUigIAHdqPXy8NQAHJLxlVU1tTx8-Q3tgkgaZLl5+IVRyNBgpIlD5CEUylw0Cyv8jULNLazj4-sHywRGPL0KtcfqQ+y5I6Nje+LxE1PTMnLzdS+LSxeXqtdqAhvYGwJAAGhBwgEtyGAA5UgAWzgiBgSWBRT+SDAdDoH3AuB+3AQIAOeEIIBEaD+dEx1HR7ExOAgKVgdDAMBwUG+OG4aSKMDA3x+zIwAE98XgMNiYDJXgjIDhkQAxAZA0gYVEAKzouARsCK8MQ6MxdA5f3ItIwXMx5ElMHVmPeMjVGpgWpwGBSQuRaQwpGppBEUF1IH1GENOpAxDegSAA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "16.12.0",
  "stylelint-config-standard": "36.0.1"
}
```

### What did you expect to happen?

I did expect no errors 

### What actually happened?

```
test.css
16:3-18 errorExpected selector "&:global(.blue)" to come before selector ".button:global(.red):hover"
    [(no-descending-specificity)](https://stylelint.io/user-guide/rules/no-descending-specificity)
```

**Comment by romainmenke** [comment]
Hi @milkyfuckinway,

[A minimal demo without nesting or `:global`](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0AjArgFwxAdvAlgM4AUSATjFAJQAEwAOjjTZADYRnw1kDmKxAJgCsQmgAZxVANyMAvo0apM2PEVIVq8ABYQAbjDJ1GzNhy69+w0QEYxEgOxDpchTiVZcBEqlZoYtBiYWCHZObj5BEXFJGRxZEAAaEAAzfFYYADkAQwBbOEQYAA9cgAd0pDBCQkTwXFSeBBBA5noQMjR0wlauZuYaVpwIAFpYQjAYHCh8HB4hwhKYMHxUpYwAT26aDHaYYxp5OJrIHHqAMQ4crIxGgCtCXBrYEurEXtbCdfTWaYxN1tYrjAPq0Ent3p8YN8cBghsd6nMMFlJlkyFA-iAARggb8QHIQLIgA)

```css
.button:is(.red) {
  color: rgb(255 0 0);
}

.button:is(.red):hover {
  color: rgb(255 100 75);
}

.button:is(.blue) {
  color: rgb(255 0 0);
}
```

This error happens because the rule compares selectors as they are the same without any pseudo classes.

So the rule uses these as context when comparing:

```css
.button { /* :is(.red) */
  color: rgb(255 0 0);
}

.button { /* :is(.red):hover */
  color: rgb(255 100 75);
}

.button { /* :is(.blue) */
  color: rgb(255 0 0);
}
```

It does calculate specificity correctly, but it is comparing things you aren't expecting to be compared. 

I think the rule is still correct however as it is possible to have this html:

```html
<button class="button blue red">...</div>
```

Such an element would match both `.button.red` and `.button.blue`.
The warning emitted by this rule is intended to caution about such cases.

-------

There is a way to rewrite your CSS so that this rule will not emit warnings:

```css
.button:global(.red) {
  color: rgb(255 0 0);
}

.button:global(.blue) {
  color: rgb(0 0 255);
}

.button:global(.red):hover {
  color: rgb(255 100 75);
}

.button:global(.blue):hover {
  color: rgb(75 100 255);
}
```

Or reduce specificity by using custom props:

```css
.button {
  --color: rgb(255 0 0);
  --color-hover: rgb(255 100 75);

  color: var(--color);
}

.button:hover {
  color: var(--color-hover);
}

.button:global(.blue) {
  --color: rgb(0 0 255);
  --color-hover: rgb(75 100 255);
}
```

I do understand that this isn't really helpful when you intend to write heavily nested CSS :)

----

**But maybe this rule predates functional pseudo classes? Should functional pseudo's be part of the context selector?**
- `.a:is(.b)` vs. `.a:is(.c)`
- `.a:not(.b)` vs. `.a:not(.c)`
- `.a:has(.b)` vs. `.a:has(.c)`
- `.a:global(.b)` vs. `.a:global(.c)`

Maybe gated behind a rule option?

**Comment by romainmenke** [comment]
Also see this related issue: https://github.com/stylelint/stylelint/issues/7844
